File: No_1448/Count Good Nodes in Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int goodNodes(TreeNode* root) {
        int ret = 0;
        DFS(root, INT_MIN, ret);
        
        return ret;
    }
    
    void DFS(TreeNode* root, int pathMax, int &ret) {
        if(root == nullptr) return;
        
        if(root->val >= pathMax) {
            pathMax = root->val;
            ret++;
        }
        
        DFS(root->left, pathMax, ret);
        DFS(root->right, pathMax, ret);
        
        return;
    }
};

File: No_1448/Count Good Nodes in Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        ans = 0
        def dfs(node, max_val):
            nonlocal ans
            if node is None:
                return
            if node.val >= max_val:
                ans += 1
            
            dfs(node.left, max(max_val, node.val))
            dfs(node.right, max(max_val, node.val))

        dfs(root, root.val)
        return ans